File: samples/dart/02reverse/reverse.dart
import "dart:io";


void main(List<String> arguments) {
    if (arguments.length < 1) {
        stderr.write("Error: Usage dart reverse.art <arg> [<arg> ...]\n");
        exit(1);
    }

    bool isfirst = true;
    for (String arg in arguments.reversed) {
        if (!isfirst) {
            stdout.write(" ");
        }
        stdout.write(arg);
        isfirst = false;
    }
    print("");
}

File: samples/java/02reverse/reverse.java
public class reverse {
    public static void main(String[] args) {
        if (args.length == 0) {
            System.err.println("Error: Usage java reverse <arg> [<arg> ...].");
            System.exit(1);
        }
        for (int i = args.length - 1; i >= 0; i--) {
            System.out.print(args[i]);
            if (i > 0) {
                System.out.print(" ");
            }
        }
        System.out.println();
    }
}
